api/v1alpha1/nnf_node_block_storage_types.go

```diff
@@ -0,0 +1,127 @@
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	dwsv1alpha2 "github.com/DataWorkflowServices/dws/api/v1alpha2"
+	"github.com/DataWorkflowServices/dws/utils/updater"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type NnfNodeBlockStorageAllocationSpec struct {
+	// Aggregate capacity of the block devices for each allocation
+	Capacity int64 `json:"capacity,omitempty"`
+
+	// List of nodes where /dev devices should be created
+	Access []string `json:"access,omitempty"`
+}
+
+// NnfNodeBlockStorageSpec defines the desired storage attributes on a NNF Node.
+// Storage spec are created on bequest of the user and fullfilled by the NNF Node Controller.
+type NnfNodeBlockStorageSpec struct {
+	// Allocations is the list of storage allocation to make
+	Allocations []NnfNodeBlockStorageAllocationSpec `json:"allocations,omitempty"`
+}
+
+type NnfNodeBlockStorageStatus struct {
+	// Allocations is the list of storage allocations that were made
+	Allocations []NnfNodeBlockStorageAllocationStatus `json:"allocations,omitempty"`
+
+	dwsv1alpha2.ResourceError `json:",inline"`
+
+	Ready bool `json:"ready"`
+}
+
+type NnfNodeBlockStorageDeviceStatus struct {
+	// NQN of the base NVMe device
+	NQN string `json:"NQN"`
+
+	// Id of the Namespace on the NVMe device (e.g., "2")
+	NamespaceId string `json:"namespaceId"`
+
+	// Total capacity allocated for the storage. This may differ from the requested storage
+	// capacity as the system may round up to the requested capacity satisify underlying
+	// storage requirements (i.e. block size / stripe size).
+	CapacityAllocated int64 `json:"capacityAllocated,omitempty"`
+}
+
+type NnfNodeBlockStorageAccessStatus struct {
+	// /dev paths for each of the block devices
+	DevicePaths []string `json:"devicePaths,omitempty"`
+
+	// Redfish ID for the storage group
+	StorageGroupId string `json:"storageGroupId,omitempty"`
+}
+
+type NnfNodeBlockStorageAllocationStatus struct {
+	Accesses map[string]NnfNodeBlockStorageAccessStatus `json:"accesses,omitempty"`
+
+	// List of NVMe namespaces used by this allocation
+	Devices []NnfNodeBlockStorageDeviceStatus `json:"devices,omitempty"`
+
+	// Total capacity allocated for the storage. This may differ from the requested storage
+	// capacity as the system may round up to the requested capacity satisify underlying
```

Comment:
stastify here too - satsifying or to satfisy?

Reply:
These were all just copied from NnfNodeStorage. I'll fix the spelling